realm/realm-library/build.gradle

```diff
@@ -46,6 +46,7 @@ repositories {
 
 dependencies {
     provided 'io.reactivex:rxjava:1.1.0'
+    compile "com.android.support:recyclerview-v7:23.1.1"
```

Comment:
I just tested locally by adding 

```
provided "com.android.support:recyclerview-v7:23.1.1"
```

And there I could build and deploy Realm without any warnings. Also following the link you provided it seems to have been fixed in 1.3. Where are you seeing that error?

In any case 11K, is far and beyond what we would could reasonable ship with the Realm package. So we would have to to find a work-around either way.